etc/scripts/json2csv.py

```diff
@@ -106,239 +75,148 @@ def load_scan(json_input):
-
-        """
-        {
-          "type": "file",
-          "name": "test_patch.py",
-          "extension": ".py",
-          "date": "2015-12-08",
-          "size": 72347,
-          "sha1": "bc1dc65b7d6b88709ce170291f93cd255bda8ffa",
-          "md5": "25edeca9fbedd5b53e7e70af0ab0140a",
-          "files_count": null,
-          "mime_type": "text/x-python",
-          "file_type": "Python script, ASCII text executable",
-          "programming_language": "Python",
-          "is_binary": null,
-          "is_text": true,
-          "is_archive": null,
-          "is_media": null,
-          "is_source": true,
-          "is_script": true
-        }
-        """
-
-        for info in entry.get('infos', []):
-            if info['type'] == 'directory':
-                if not location.endswith('/'):
-                    location = location + '/'
-            inf = OrderedDict(Resource=location)
-            for k, val in info.items():
-                key = 'info_' + k
-                inf[key] = val
-                keys.add(key)
-            rows.append(inf)
-
-        """
-      "licenses": [
-        {
-          "key": "apache-2.0",
-          "score": 100.0,
-          "short_name": "Apache 2.0",
-          "category": "Attribution",
-          "owner": "Apache Software Foundation",
-          "homepage_url": "http://www.apache.org/licenses/",
-          "text_url": "http://www.apache.org/licenses/LICENSE-2.0",
-          "dejacode_url": "https://enterprise.dejacode.com/license_library/Demo/apache-2.0/",
-          "spdx_license_key": "Apache-2.0",
-          "spdx_url": "http://spdx.org/licenses/Apache-2.0",
-          "start_line": 4,
-          "end_line": 23
-        },
-        {
-          "key": "scancode-acknowledgment",
-          "score": 100.0,
-          "short_name": "ScanCode acknowledgment",
-          "category": "Attribution",
-          "owner": "nexB Inc.",
-          "homepage_url": "https://github.com/nexB/scancode-toolkit/",
-          "text_url": "",
-          "dejacode_url": "https://enterprise.dejacode.com/license_library/Demo/scancode-acknowledgment/",
-          "spdx_license_key": "",
-          "spdx_url": "",
-          "start_line": 4,
-          "end_line": 23
-        }
-      ],
-        """
-        for licensing in entry.get('licenses', []):
-            lic = OrderedDict(Resource=location)
+
+    for scanned_file in scan:
+        path = scanned_file['path']
+        path = path if path.startswith('/') else '/' + path
+
+        if scanned_file.get('type', '') == 'directory':
+            if not path.endswith('/'):
+                path = path + '/'
+
+        path = '/code' + path
+
+        file_info = OrderedDict(Resource=path)
+        info_details = OrderedDict(((k, v) for k, v in scanned_file.items() if k != 'path' and not isinstance(v, list)))
+        file_info.update(info_details)
+        yield file_info
+
+        error_info = OrderedDict(Resource=path)
```

Comment:
Since there is only one scan_errors attribute for any given file, why not yielding this with the file_info above?